lib/std/static_string_map.zig

```diff
@@ -1,446 +1,170 @@
-        const empty_keys = [0][]const u8{};
-        const empty_vals = [0]V{};
-
-        /// Returns a map backed by static, comptime allocated memory.
-        ///
-        /// `kvs_list` must be either a list of `struct { []const u8, V }`
-        /// (key-value pair) tuples, or a list of `struct { []const u8 }`
-        /// (only keys) tuples if `V` is `void`.
-        pub inline fn initComptime(comptime kvs_list: anytype) Self {
-            comptime {
-                var self = Self{};
-                if (kvs_list.len == 0)
-                    return self;
-
-                // Since the KVs are sorted, a linearly-growing bound will never
-                // be sufficient for extreme cases. So we grow proportional to
-                // N*log2(N).
-                @setEvalBranchQuota(10 * kvs_list.len * std.math.log2_int_ceil(usize, kvs_list.len));
+        kvs: []const Kv,
 
-                var sorted_keys: [kvs_list.len][]const u8 = undefined;
-                var sorted_vals: [kvs_list.len]V = undefined;
-
-                self.initSortedKVs(kvs_list, &sorted_keys, &sorted_vals);
-                const final_keys = sorted_keys;
-                const final_vals = sorted_vals;
-                self.kvs = &.{
-                    .keys = &final_keys,
-                    .values = &final_vals,
-                    .len = @intCast(kvs_list.len),
-                };
-
-                var len_indexes: [self.max_len + 1]u32 = undefined;
-                self.initLenIndexes(&len_indexes);
-                const final_len_indexes = len_indexes;
-                self.len_indexes = &final_len_indexes;
-                self.len_indexes_len = @intCast(len_indexes.len);
-                return self;
-            }
-        }
-
-        /// Returns a map backed by memory allocated with `allocator`.
-        ///
-        /// Handles `kvs_list` the same way as `initComptime()`.
-        pub fn init(kvs_list: anytype, allocator: mem.Allocator) !Self {
-            var self = Self{};
-            if (kvs_list.len == 0)
-                return self;
-
-            const sorted_keys = try allocator.alloc([]const u8, kvs_list.len);
-            errdefer allocator.free(sorted_keys);
-            const sorted_vals = try allocator.alloc(V, kvs_list.len);
-            errdefer allocator.free(sorted_vals);
-            const kvs = try allocator.create(KVs);
-            errdefer allocator.destroy(kvs);
-
-            self.initSortedKVs(kvs_list, sorted_keys, sorted_vals);
-            kvs.* = .{
-                .keys = sorted_keys.ptr,
-                .values = sorted_vals.ptr,
-                .len = @intCast(kvs_list.len),
+        /// Initializes the map at comptime
+        pub fn initComptime(comptime kvs_list: anytype) Self {
+            const kvs: []const Kv = comptime blk: {
+                var kv_list: []const Kv = &.{};
+                for (kvs_list) |kv| {
+                    kv_list = kv_list ++ .{.{ .key = kv[0], .value = kv[1] }};
```

Comment:
I think you should be accessing kvs with the `kv.@"0"` and `kv.@"1"` syntax.

Reply:
I forget why this syntax was used.  I think I remember @squeek502 suggested it in some review.

Reply:
Good memory: https://github.com/ziglang/zig/pull/5359#discussion_r430750998

In particular, it was necessary to get this test from the original ComptimeStringMap PR (https://github.com/ziglang/zig/pull/5452) working:

```zig
test "ComptimeStringMap slice of structs" {
    const KV = struct {
        @"0": []const u8,
        @"1": TestEnum,
    };
    const slice: []const KV = &[_]KV{
        .{.@"0" = "these", .@"1" = .D},
        .{.@"0" = "have", .@"1" = .A},
        .{.@"0" = "nothing", .@"1" = .B},
        .{.@"0" = "incommon", .@"1" = .C},
        .{.@"0" = "samelen", .@"1" = .E},
    };
    const map = ComptimeStringMap(TestEnum, slice);

    testMap(map);
}
```

It looks like the need for this was removed when proper tuple types were added, though (the equivalent tests in `StaticStringMap` use `struct { []const u8, TestEnum };`), so `[0]` and `[1]` might be fine now.

Reply:
Ah thanks for clearing that up.  Please ignore my suggestion then.